src/libcore/num/mod.rs

```diff
@@ -110,6 +111,57 @@ assert_eq!(size_of::<Option<core::num::", stringify!($Ty), ">>(), size_of::<", s
                 }
             }
 
+            #[stable(feature = "nonzero_bitor", since = "1.43.0")]
+            impl BitOr for $Ty {
+                type Output = Self;
+                #[inline]
+                fn bitor(self, rhs: Self) -> Self::Output {
+                    // Safety: since `self` and `rhs` are both nonzero, the
```

Comment:
I remember hearing that libcore was going to require safety annotations, so I added them even though they're trivial. I can remove them (or remove the duplicates) if it's extraneous or noisy.

I also used new_unchecked instead of just doing `$Ty(...)` (despite it being in scope) to call attention to the unsafety but similarly don't have strong opinions here.